node/network/bridge/src/network.rs

```diff
@@ -135,105 +106,69 @@ pub trait Network: Clone + Send + 'static {
 	);
 
 	/// Report a given peer as either beneficial (+) or costly (-) according to the given scalar.
-	fn report_peer(
-		&mut self,
-		who: PeerId,
-		cost_benefit: Rep,
-	) -> BoxFuture<SubsystemResult<()>> {
-		async move {
-			self.action_sink()
-				.send(NetworkAction::ReputationChange(who, cost_benefit))
-				.await
-		}
-		.boxed()
-	}
+	fn report_peer(&self, who: PeerId, cost_benefit: Rep);
 
 	/// Disconnect a given peer from the peer set specified without harming reputation.
-	fn disconnect_peer(
-		&mut self,
-		who: PeerId,
-		peer_set: PeerSet,
-	) -> BoxFuture<SubsystemResult<()>> {
-		async move {
-			self.action_sink()
-				.send(NetworkAction::DisconnectPeer(who, peer_set))
-				.await
-		}
-		.boxed()
-	}
+	fn disconnect_peer(&self, who: PeerId, peer_set: PeerSet);
 
 	/// Write a notification to a peer on the given peer-set's protocol.
 	fn write_notification(
-		&mut self,
+		&self,
 		who: PeerId,
 		peer_set: PeerSet,
 		message: Vec<u8>,
-	) -> BoxFuture<SubsystemResult<()>> {
-		async move {
-			self.action_sink()
-				.send(NetworkAction::WriteNotification(who, peer_set, message))
-				.await
-		}
-		.boxed()
-	}
+	);
 }
 
 #[async_trait]
 impl Network for Arc<NetworkService<Block, Hash>> {
 	fn event_stream(&mut self) -> BoxStream<'static, NetworkEvent> {
-		NetworkService::event_stream(self, "polkadot-network-bridge").boxed()
+		NetworkService::event_stream(self, "polkadot-network-bridge").filter_map(|e| async move {
+			match e {
+				NetworkEvent::NotificationStreamOpened { .. }
+				| NetworkEvent::NotificationStreamClosed { .. }
+				| NetworkEvent::NotificationsReceived { .. } => Some(e),
+				_ => None,
```

Comment:
Not sure about this. If we want to pre-filter events, we should probably translate them into another type with only the constructors we are interested in. Otherwise, if we ever get interested in other events, the developer in question will likely just start handling them in lib.rs and we will all start debugging why it does not work.

Reply:
I agree. My thinking was that we we'd spend CPU cycles in `handle_network_messages` waking up on unrelevant events, so it'd be better to filter these out at the stream level. I'll add some types here.

Reply:
I fail to see the difference. Like the `handle_network_messages` will just ignore those messages, the same as we do with filtering. I don't see how this saves us anything.

Reply:
I thought `filter_map` would be a bit different than a CPU loop
https://docs.rs/futures-util/0.3.15/src/futures_util/stream/stream/filter_map.rs.html#61-98, but I guess you're right. I'll revert this change.